File: android/app/src/main/java/com/talentlens/app/ui/screens/ResultScreen.kt
package com.talentlens.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.talentlens.app.model.AssessmentResult
import com.talentlens.app.ui.components.CertificateDialog
import com.talentlens.app.ui.theme.*

@Composable
fun ResultScreen(
    assessment: AssessmentResult,
    onRetest: () -> Unit,
    onNavigateToScout: () -> Unit
) {
    var showCertificate by remember { mutableStateOf(false) }
    var isBroadcasted by remember { mutableStateOf(false) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(BackgroundDark)
            .padding(16.dp)
            .verticalScroll(rememberScrollState()),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Top Banner
        Surface(
            color = VerifiedEmerald.copy(alpha = 0.15f),
            shape = RoundedCornerShape(20.dp),
            border = androidx.compose.foundation.BorderStroke(1.dp, VerifiedEmerald.copy(alpha = 0.4f)),
            modifier = Modifier.fillMaxWidth()
        ) {
            Row(
                modifier = Modifier.padding(16.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                Icon(Icons.Default.VerifiedUser, contentDescription = null, tint = VerifiedEmerald, modifier = Modifier.size(28.dp))
                Column {
                    Text("AI VERIFIED ASSESSMENT", color = TextPrimary, fontSize = 12.sp, fontWeight = FontWeight.Bold, fontFamily = FontFamily.Monospace)
                    Text("Hash: ${assessment.verificationHash}", color = VerifiedEmerald, fontSize = 11.sp, fontFamily = FontFamily.Monospace)
                }
            }
        }

        // Main Result Card
        Card(
            shape = RoundedCornerShape(24.dp),
            colors = CardDefaults.cardColors(containerColor = CardBackground),
            modifier = Modifier
                .fillMaxWidth()
                .border(1.dp, CardBorder, RoundedCornerShape(24.dp))
        ) {
            Column(modifier = Modifier.padding(20.dp), verticalArrangement = Arrangement.spacedBy(16.dp)) {
                
                // Athlete details
                Text(assessment.exerciseType.title.uppercase(), color = BrandOrange, fontSize = 11.sp, fontWeight = FontWeight.Bold)
                Text(assessment.athleteName, color = TextPrimary, fontSize = 24.sp, fontWeight = FontWeight.Black)
                Text("${assessment.district}, ${assessment.state} • ${assessment.age}y • ${assessment.sport.displayName}", color = TextSecondary, fontSize = 12.sp)

                Divider(color = CardBorder)

                // 3-Metric KPI Grid
                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text("SCORE", color = TextSecondary, fontSize = 10.sp, fontWeight = FontWeight.Bold)
                        Text("${assessment.score}", color = TextPrimary, fontSize = 32.sp, fontWeight = FontWeight.Black)
                        Text(assessment.exerciseType.metricUnit.uppercase(), color = BrandOrange, fontSize = 10.sp, fontWeight = FontWeight.Bold)
                    }
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text("NATIONAL %ILE", color = BrandOrange, fontSize = 10.sp, fontWeight = FontWeight.Bold)
                        Text("${assessment.percentile.toInt()}%", color = BrandOrange, fontSize = 32.sp, fontWeight = FontWeight.Black)
                        Text("SAI Standards", color = TextSecondary, fontSize = 10.sp)
                    }
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text("FORM SCORE", color = VerifiedEmerald, fontSize = 10.sp, fontWeight = FontWeight.Bold)
                        Text("${assessment.biomechanics.formScore}%", color = VerifiedEmerald, fontSize = 32.sp, fontWeight = FontWeight.Black)
                        Text("Precision", color = TextSecondary, fontSize = 10.sp)
                    }
                }

                // Talent Tier Badge
                Surface(
                    color = Color(assessment.talentTier.badgeColorHex).copy(alpha = 0.15f),
                    shape = RoundedCornerShape(12.dp),
                    border = androidx.compose.foundation.BorderStroke(1.dp, Color(assessment.talentTier.badgeColorHex).copy(alpha = 0.4f)),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text(
                        text = assessment.talentTier.displayName,
                        color = Color(assessment.talentTier.badgeColorHex),
                        fontSize = 13.sp,
                        fontWeight = FontWeight.Bold,
                        textAlign = TextAlign.Center,
                        modifier = Modifier.padding(vertical = 8.dp)
                    )
                }
            }
        }

        // Action Buttons
        Button(
            onClick = { showCertificate = true },
            colors = ButtonDefaults.buttonColors(containerColor = CardBackground),
            shape = RoundedCornerShape(16.dp),
            modifier = Modifier.fillMaxWidth().height(52.dp).border(1.dp, CardBorder, RoundedCornerShape(16.dp))
        ) {
            Icon(Icons.Default.WorkspacePremium, contentDescription = null, tint = BrandOrange)
            Spacer(Modifier.width(8.dp))
            Text("View & Print Official Certificate", color = TextPrimary, fontWeight = FontWeight.Bold)
        }

        Button(
            onClick = { isBroadcasted = true },
            enabled = !isBroadcasted,
            colors = ButtonDefaults.buttonColors(containerColor = BrandOrange),
            shape = RoundedCornerShape(16.dp),
            modifier = Modifier.fillMaxWidth().height(52.dp)
        ) {
            Icon(Icons.Default.Send, contentDescription = null)
            Spacer(Modifier.width(8.dp))
            Text(if (isBroadcasted) "Broadcasted to Live Scouts ✓" else "Push to Live Scout Network", fontWeight = FontWeight.Bold)
        }

        OutlinedButton(
            onClick = onRetest,
            shape = RoundedCornerShape(16.dp),
            modifier = Modifier.fillMaxWidth().height(52.dp)
        ) {
            Icon(Icons.Default.Refresh, contentDescription = null, tint = TextSecondary)
            Spacer(Modifier.width(8.dp))
            Text("Retest / Switch Exercise", color = TextSecondary)
        }
    }

    if (showCertificate) {
        CertificateDialog(
            assessment = assessment,
            onDismiss = { showCertificate = false }
        )
    }
}

File: ios/TalentLens/UI/Screens/ResultScreen.swift
//
//  ResultScreen.swift
//  TalentLens
//
//  Created for TalentLens Olympic-Grade Athletic Testing Suite.
//

import SwiftUI

public struct ResultScreen: View {
    public let assessment: AssessmentResult
    public let onRetest: () -> Void
    public let onNavigateToScout: () -> Void
    
    @State private var showCertificate: Bool = false
    @State private var isBroadcasted: Bool = false
    
    public init(
        assessment: AssessmentResult,
        onRetest: @escaping () -> Void,
        onNavigateToScout: @escaping () -> Void
    ) {
        self.assessment = assessment
        self.onRetest = onRetest
        self.onNavigateToScout = onNavigateToScout
    }
    
    private var sportProfile: SportTrainingProfile {
        SportTrainingDatabase.profile(for: assessment.sport)
    }
    
    private var drillInfo: SportTrainingDrill? {
        sportProfile.recommendedDrills.first(where: { $0.exerciseType == assessment.exerciseType })
    }
    
    public var body: some View {
        ZStack {
            TLTheme.backgroundDark.edgesIgnoringSafeArea(.all)
            
            ScrollView {
                VStack(spacing: 16) {
                    // Top Banner
                    HStack(spacing: 12) {
                        Image(systemName: "checkmark.seal.fill")
                            .foregroundColor(TLTheme.verifiedEmerald)
                            .font(.title2)
                        
                        VStack(alignment: .leading, spacing: 2) {
                            Text("AI VERIFIED ASSESSMENT")
                                .font(.system(size: 12, weight: .bold, design: .monospaced))
                                .foregroundColor(TLTheme.textPrimary)
                            Text("Hash: \(assessment.verificationHash)")
                                .font(.system(size: 11, design: .monospaced))
                                .foregroundColor(TLTheme.verifiedEmerald)
                        }
                        Spacer()
                    }
                    .padding(16)
                    .background(TLTheme.verifiedEmerald.opacity(0.15))
                    .cornerRadius(20)
                    .overlay(
                        RoundedRectangle(cornerRadius: 20)
                            .stroke(TLTheme.verifiedEmerald.opacity(0.4), lineWidth: 1)
                    )
                    .padding(.horizontal)
                    .padding(.top, 16)
                    
                    // Main Result Card
                    VStack(alignment: .leading, spacing: 16) {
                        VStack(alignment: .leading, spacing: 4) {
                            HStack {
                                Text(assessment.exerciseType.title.uppercased())
                                    .font(.system(size: 11, weight: .bold))
                                    .foregroundColor(TLTheme.brandOrange)
                                Spacer()
                                Text("\(assessment.sport.iconEmoji) \(assessment.sport.displayName)")
                                    .font(.system(size: 11, weight: .bold))
                                    .foregroundColor(TLTheme.cyberCyan)
                            }
                            
                            Text(assessment.athleteName)
                                .font(.system(size: 24, weight: .black))
                                .foregroundColor(TLTheme.textPrimary)
                            
                            Text("\(assessment.district), \(assessment.state) • \(assessment.age)y • \(assessment.sport.displayName)")
                                .font(.system(size: 12))
                                .foregroundColor(TLTheme.textSecondary)
                        }
                        
                        Divider().background(TLTheme.cardBorder)
                        
                        // 3-Metric KPI Grid
                        HStack {
                            VStack(spacing: 4) {
                                Text("SCORE")
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(TLTheme.textSecondary)
                                Text("\(assessment.score)")
                                    .font(.system(size: 32, weight: .black))
                                    .foregroundColor(TLTheme.textPrimary)
                                Text(assessment.exerciseType.metricUnit.uppercased())
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(TLTheme.brandOrange)
                            }
                            .frame(maxWidth: .infinity)
                            
                            VStack(spacing: 4) {
                                Text("NATIONAL %ILE")
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(TLTheme.brandOrange)
                                Text("\(Int(assessment.percentile))%")
                                    .font(.system(size: 32, weight: .black))
                                    .foregroundColor(TLTheme.brandOrange)
                                Text("SAI Standards")
                                    .font(.system(size: 10))
                                    .foregroundColor(TLTheme.textSecondary)
                            }
                            .frame(maxWidth: .infinity)
                            
                            VStack(spacing: 4) {
                                Text("FORM SCORE")
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(TLTheme.verifiedEmerald)
                                Text("\(assessment.biomechanics.formScore)%")
                                    .font(.system(size: 32, weight: .black))
                                    .foregroundColor(TLTheme.verifiedEmerald)
                                Text("Precision")
                                    .font(.system(size: 10))
                                    .foregroundColor(TLTheme.textSecondary)
                            }
                            .frame(maxWidth: .infinity)
                        }
                        
                        // Sport Readiness Context Card
                        if let drill = drillInfo {
                            VStack(alignment: .leading, spacing: 6) {
                                HStack {
                                    Text("SPORT SPECIFIC READINESS")
                                        .font(.system(size: 10, weight: .bold, design: .monospaced))
                                        .foregroundColor(TLTheme.cyberCyan)
                                    Spacer()
                                    Text(drill.importanceTier)
                                        .font(.system(size: 9, weight: .bold))
                                        .foregroundColor(TLTheme.brandOrange)
                                }
                                
                                Text("Why it matters: \(drill.roleRationale)")
                                    .font(.system(size: 11))
                                    .foregroundColor(TLTheme.textPrimary)
                                
                                Text("Gym Standard: \(drill.gymTargetScore)")
                                    .font(.system(size: 10, weight: .bold, design: .monospaced))
                                    .foregroundColor(TLTheme.textSecondary)
                            }
                            .padding(12)
                            .background(TLTheme.cardBackground.opacity(0.8))
                            .cornerRadius(12)
                            .overlay(RoundedRectangle(cornerRadius: 12).stroke(TLTheme.cardBorder, lineWidth: 1))
                        }
                        
                        // Talent Tier Badge
                        Text(assessment.talentTier.displayName)
                            .font(.system(size: 13, weight: .bold))
                            .foregroundColor(Color(hex: assessment.talentTier.badgeColorHex))
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 10)
                            .background(Color(hex: assessment.talentTier.badgeColorHex).opacity(0.15))
                            .cornerRadius(12)
                            .overlay(
                                RoundedRectangle(cornerRadius: 12)
                                    .stroke(Color(hex: assessment.talentTier.badgeColorHex).opacity(0.4), lineWidth: 1)
                            )
                    }
                    .padding(20)
                    .background(TLTheme.cardBackground)
                    .cornerRadius(24)
                    .overlay(
                        RoundedRectangle(cornerRadius: 24)
                            .stroke(TLTheme.cardBorder, lineWidth: 1)
                    )
                    .padding(.horizontal)
                    
                    // Actions
                    VStack(spacing: 12) {
                        Button(action: { showCertificate = true }) {
                            HStack(spacing: 8) {
                                Image(systemName: "rosette")
                                    .foregroundColor(TLTheme.brandOrange)
                                Text("View & Print Official Certificate")
                                    .font(.system(size: 15, weight: .bold))
                                    .foregroundColor(TLTheme.textPrimary)
                            }
                            .frame(maxWidth: .infinity)
                            .frame(height: 52)
                            .background(TLTheme.cardBackground)
                            .cornerRadius(16)
                            .overlay(RoundedRectangle(cornerRadius: 16).stroke(TLTheme.cardBorder, lineWidth: 1))
                        }
                        
                        Button(action: {
                            isBroadcasted = true
                        }) {
                            HStack(spacing: 8) {
                                Image(systemName: isBroadcasted ? "checkmark" : "paperplane.fill")
                                Text(isBroadcasted ? "Broadcasted to Live Scouts ✓" : "Push to Live Scout Network")
                                    .font(.system(size: 15, weight: .bold))
                            }
                            .frame(maxWidth: .infinity)
                            .frame(height: 52)
                            .background(TLTheme.brandOrange)
                            .foregroundColor(.white)
                            .cornerRadius(16)
                        }
                        .disabled(isBroadcasted)
                        
                        Button(action: onRetest) {
                            HStack(spacing: 8) {
                                Image(systemName: "arrow.clockwise")
                                    .foregroundColor(TLTheme.textSecondary)
                                Text("Retest / Switch Exercise Drill")
                                    .font(.system(size: 14))
                                    .foregroundColor(TLTheme.textSecondary)
                            }
                            .frame(maxWidth: .infinity)
                            .frame(height: 50)
                            .background(Color.clear)
                            .overlay(RoundedRectangle(cornerRadius: 16).stroke(TLTheme.cardBorder, lineWidth: 1))
                        }
                    }
                    .padding(.horizontal)
                    .padding(.bottom, 24)
                }
            }
        }
        .sheet(isPresented: $showCertificate) {
            CertificateModalView(
                assessment: assessment,
                onDismiss: { showCertificate = false }
            )
        }
    }
}
